### DISCUSSION TITLE
How to add Global Variable That can be access to everywhere in the project

### DISCUSSION THREAD

**Discussion by naveensemidot** [adonisjs/core#3966] [Help]
Hello,

Is there any way, I can add a Custom variable or a function to access anywhere in the project.

Refernce Code

For Custom Variable
Simillar to this
`export const fileTypes = {
  CSS: "text/css",
  CSV: "text/csv",
  GIF: "image/gif",
  HTML: "text/html",
  };`
  
  Let Suppose in the validator.ts, I required to access _fileTypes.CSV_

  ------------------------------------------------------------------------
For Member Function  
`export const replaceAllOccurance = (string, toReplace, withReplace) => {
  return string.split(toReplace).join(withReplace);
};`

And in the Controller or Model
I want to call _replaceAllOccurance(param1,param2,param3);_
  
How can we do in the AdonisJS 5.

**Comment by RomainLanz** [comment]
Hey @naveensemidot! 👋🏻 

There is nothing special to be done. AdonisJS is like any other Node.js application.
Export the variable/function when defining it and import it in the other file.

```ts
// file: foo.ts
export const foo = 'acme'
```

```ts
// file: bar.ts
import { foo } from './foo'

console.log(foo) // acme
```

📚 [Modules, introduction](https://javascript.info/modules-intro)
